content/blog/prompt-engg.md

```diff
@@ -0,0 +1,143 @@
++++
+title = "Prompt Engineering 101"
+slug = "prompt-engg"
+date = 2023-06-12T00:10:46+05:30
+image = "/images/2023/prompt-engg.png"
+draft = false
+authors = ["Abhishek Keshri"]
+description = ""
+tags = ["AI", "Learnings", "Software Craftsmanship"]
+categories = ["AI", "Learnings", "Software Craftsmanship"]
+type = ""
++++
+
+Are you feeling lost in the new world of AI tools and Prompt Engineering?
+Don't worry, we got you!
+In this article, we'll demystify everything and equip you with the knowledge you need to get the most out of AI tools like ChatGPT.
+So, let's get started!
+
+# What Is Prompt Engineering?
+
+Prompt engineering is like being the conductor of an AI orchestra.
+It's the art of crafting carefully tailored instructions, or prompts, that guide language models like GPT-3.5 (Used in ChatGPT) to produce the outputs we desire.
+Think of it as giving your AI companion the perfect set of directions to create the results you're looking for.
+
+In this article, we'll explore the fundamental principles of prompt engineering and how they can be applied in practice.
+
+## Principle 1: Write clear and specific instructions
+
+The clarity and specificity of instructions greatly influence the output generated by the model.
+Here are some guidelines to follow:
+
+1. Use delimiters to clearly indicate distinct parts of the input.
+   Delimiters can be anything like triple backticks (```) or semicolon (;).
+   For example, when providing multiple inputs, you can separate them using delimiters to ensure the model processes them correctly.
+
+___
+Input 1: This is the first input;
+
+Input 2: This is the second input;
+___
+
+2. Ask for a structured output format such as JSON or HTML.
+   This helps in obtaining well-organized and easily parseable results.
+   For instance, if you want the model to generate an HTML table, specify the desired structure in your prompt.
+
+3. Instruct the model to check whether certain conditions are satisfied.
+   This allows you to incorporate logical checks in the output.
+   For example, you can ask the model to verify if a given statement is true or false based on the provided information.
+
+4. Utilize the power of "few-shot" prompting.
+   Instead of asking the model to perform a task from scratch, provide an example or a partial solution and ask it to do something similar.
+   This enables the model to generalize from the given example and produce the desired output.
+
```

Comment:
Here too see if you can give a specific example for this point. 
Could be a prompt with an example and asking chatgpt to generate something similar.